glean/src/server_events.rs

```diff
@@ -0,0 +1,308 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Required imports
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+use serde_json;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+// log type string used to identify logs to process in the Moz Data Pipeline
+const GLEAN_EVENT_MOZLOG_TYPE: &str = "glean-server-event";
+
+// Code below is static, regardless of what is defined in `metrics.yaml`:
+pub struct GleanEventsLogger {
```

Comment:
Since we specify a docstring for `GleanEventsLogger::record()` let's add one here to explain what's special about this struct.